src/test/java/com/iexec/core/task/update/TaskUpdateRequestManagerTests.java

```diff
@@ -63,21 +61,19 @@ void shouldNotPublishRequestSinceItemAlreadyAdded() throws ExecutionException, I
                 buildTaskUpdate(CHAIN_TASK_ID, null, null, null)
         );
 
-        CompletableFuture<Boolean> booleanCompletableFuture = taskUpdateRequestManager.publishRequest(CHAIN_TASK_ID);
-        booleanCompletableFuture.join();
+        boolean booleanCompletableFuture = taskUpdateRequestManager.publishRequest(CHAIN_TASK_ID);
 
-        Assertions.assertThat(booleanCompletableFuture.get()).isFalse();
+        Assertions.assertThat(booleanCompletableFuture).isFalse();
     }
 
     @Test
     void shouldNotPublishRequestSinceTaskDoesNotExist() throws ExecutionException, InterruptedException {
         when(taskService.getTaskByChainTaskId(CHAIN_TASK_ID))
                 .thenReturn(Optional.empty());
 
-        CompletableFuture<Boolean> booleanCompletableFuture = taskUpdateRequestManager.publishRequest(CHAIN_TASK_ID);
-        booleanCompletableFuture.join();
+        boolean booleanCompletableFuture = taskUpdateRequestManager.publishRequest(CHAIN_TASK_ID);
 
-        Assertions.assertThat(booleanCompletableFuture.get()).isFalse();
+        Assertions.assertThat(booleanCompletableFuture).isFalse();
```

Comment:
The variable name becomes a bit weird as the method does not return a `CompletableFuture` in the new implementation.
```suggestion
        Assertions.assertThat(publishRequestStatus).isFalse();
```

If we go for a fix, it will have to be done in several test methods.

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-core/pull/633/commits/76906fa5718bf404f8b62565726763503c36bf06.